### ISSUE TITLE
Song titles, album etc non readable characters

### ISSUE THREAD

**Issue by huhas12** [owncloud/music#924]
![2021-11-29_9-06-49](https://user-images.githubusercontent.com/65531798/143823275-cdb4628f-b235-49ab-9705-4a4dc332f101.jpg)

The lunguage of the song titles, artist etc is in Greek and is not displayed properly. Any ideas how to solve it?

**Comment by paulijar** [comment]
The likely explanation is that your files use an invalid combination of metadata tag type and character encoding: The most common mistake like this is that ID3v1 tags are filled with encoding other than ISO-8859-1. In case of of Greece text, that would likely be ISO-8859-7. But ID3v1 is actually specified to always use ISO-8859-1 and anything else is abuse of those tags. Historically, this was the only way to write non-Latin metadata to MP3 files but this hasn't been the case for the last ~20 years.

Now, if this is the case, then the proper way to solve it would be to use ID3v2.x tags instead (or in addition to) ID3v1, which would enable using the UTF-8 encoding. This is basically the only character encoding which anyone should be using these days, as it enables compatible representation of just about any scripture the mankind has ever invented.

All that being said, it is also possible that your files already contain (also) the UTF-8-encoded metadata tags, but in combination with some exotic legacy tags which then confuses the getID3 library we use for the metadata extraction. So the question is, do you know what kind of tags there are in your files? If you want, you can also send one such file to me as a sample, and I can take a look. You can find my email address from my github profile.

**Comment by huhas12** [comment]
Hello Paulijar, 

Thank you forthe info. I am not familiar with meta tags. I 've already sent you an email with links to some of the files. 

Don't know if it's beyond the scope of the project, but is there any way to mass edit meta tags and encodings of the files?

**Comment by paulijar** [comment]
Okay, I got and looked at the six sample files, three of which show correct Greece alphabet and three don't. All the three files with corrupted metadata have both ID3v1 and ID3v2.3 tags, but both tag types have the same characters encoding issue; as a result, none of the applications I tested could show anything but corrupted metadata for these files.

Of the three "correctly" working file, two actually didn't contain any kind of metadata. In that case, the Music app makes the best guess of using the file and folder names as track, album, and artist titles. Finally, one of the six sample files had correct ID3v2.3 metadata with no character encoding issues.

For all my own mass-tagging needs, I have used the [Mp3tag](https://www.mp3tag.de/en/index.html) software. It's very powerful but requires some practice and investigation to use to its full extent. But even if you skip the more advanced features, it will still be easier to manipulate a large set of files with it than with most alternatives.
